bc/core/utils/images.py

```diff
@@ -278,3 +278,163 @@ def to_bytes(self) -> bytes:
 
         # Turn the BytesIO object back into a bytes object
         return buffer.getvalue()
+
+
+@dataclass
+class SponsoredThumbnail:
+    title: str
+    thumbnail: bytes
+    format: str = "png"
+    margin: int = 40
+    small_text: str | None = None
+    title_font_path: str | None = finders.find("fonts/CooperHewitt-Bold.otf")
+    small_font_path: str | None = finders.find("fonts/CooperHewitt-Medium.otf")
+    text_box: Image = field(init=False)
+    background: Image = field(init=False)
+    overlay_layer: Image = field(init=False)
+
+    def __post_init__(self) -> None:
+        self.title_font = ImageFont.truetype(self.title_font_path, 46)
+        self.small_font = ImageFont.truetype(self.small_font_path, 22)
+        self.background = Image.open(io.BytesIO(self.thumbnail)).convert(
+            "RGBA"
+        )
+        # make a transparent image for the text
+        self.overlay_layer = Image.new(
+            "RGBA", self.background.size, (255, 255, 255, 0)
+        )
+        if not self.small_text:
+            self.small_text = "Learn more about supporting Free Law Project at https://bots.law/sponsors/"
+
+    def get_text_box_dimensions(self) -> tuple[int, int]:
+        """
+        Returns the dimensions(width and height, in pixels) of a text box to render
+        the sponsored text.
+
+        Returns:
+            tuple[int,int]: (width, height) dimensions
+        """
+        title_w, title_h = self.title_font.getbbox(self.title)[-2:]
+        small_w, small_h = self.small_font.getbbox(self.small_text)[-2:]
+
+        bbox_height = title_h + small_h
+        bbox_width = title_w if title_w >= small_w else small_w
```

Comment:
```suggestion
        bbox_width = max(title_w, small_w)
```

Reply:
You're right. I'll use this built-in function instead.